tests/Fixer/Whitespace/BlankLineBeforeControlStatementFixerTest.php

```diff
@@ -0,0 +1,1046 @@
+while (true) {
+    while (true) {
+        $foo = true;
+        continue 2;
+    }
+}',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithDeclare
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDeclare($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'declare',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDeclare()
+    {
+        return array(
+            array(
+'<?php
+declare(ticks=1);',
+            ),
+            array(
+'<?php
+
+$foo = "bar";
+
+declare(ticks=1);',
+'<?php
+
+$foo = "bar";
+declare(ticks=1);',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithDo
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithDo($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'do',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithDo()
+    {
+        return array(
+            array(
+'<?php
+do {
+} while (true);',
+            ),
+            array(
+'<?php
+$foo = "bar";
+
+do {
+} while (true);',
+'<?php
+$foo = "bar";
+do {
+} while (true);',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithElse
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithElse($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'else',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithElse()
+    {
+        return array(
+            array(
+'<?php
+if (true) {
+    echo $b;
+} 
+
+else {
+    echo $c;
+}',
+            ),
+            array(
+'<?php
+if (true) {
+    echo $b;
+}
+
+else {
+    echo $c;
+}',
+'<?php
+if (true) {
+    echo $b;
+} else {
+    echo $c;
+}',
+            ),
+            array(
+'<?php
+if (true) 
+    echo $b;
+    
+else 
+    echo $c;',
+            ),
+            array(
+'<?php
+if (true) 
+    echo $b;
+
+else 
+    echo $c;',
+'<?php
+if (true) 
+    echo $b;
+else 
+    echo $c;',
+            ),
+        );
+    }
+
+    /**
+     * @dataProvider providerFixWithElseIf
+     *
+     * @param string      $expected
+     * @param string|null $input
+     */
+    public function testFixWithElseIf($expected, $input = null)
+    {
+        $this->fixer->configure(array(
+            'elseif',
+        ));
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return array
+     */
+    public function providerFixWithElseIf()
+    {
+        return array(
+            array(
+'<?php
+if ($c) {
+    echo $b;
+} 
+
+elseif ($b) {
```

Comment:
if you mean fixing by removing the empty line than I would suggest don't bother, we've a fixer for that ;)